api/models/Centres.js

```diff
@@ -90,18 +90,24 @@ const model = {
       return response;
     }
   },
+
   getGenderAndCentreByCIDLocation: function (location) {
-    return this.findOne(
-      {
-        or: [
-          {male_cid_name: location},
-          {female_cid_name: location}
-        ]
-      })
-      .then(centre => centre ? {
-        centre: centre.id,
-        gender: centre.male_cid_name === location ? "male" : "female"
-      } : centre);
+    return this.find().then(centres =>
+      _.compact(_.map(centres, centre => {
```

Comment:
if you make a JIRA ticket for that ^^ as a future improvement I'd be happy to merge this now if that helps since it's not going to be too expensive for now?